src/nvim/getchar.c

```diff
@@ -1687,31 +1754,7 @@ static int vgetorpeek(int advance)
           os_breakcheck();                      /* check for CTRL-C */
         keylen = 0;
         if (got_int) {
-          /* flush all input */
-          c = inchar(typebuf.tb_buf, typebuf.tb_buflen - 1, 0L,
```

Comment:
`flush_all_input(typebuf)` could be a reusable function: it mutates state, but has clear semantics.